moveit_ros/moveit_servo/CMakeLists.txt

```diff
@@ -1,146 +1,188 @@
-  ${LIBRARY_NAME}
-  ${catkin_LIBRARIES}
-  ${Eigen_LIBRARIES}
-  ${Boost_LIBRARIES}
+set_target_properties(${LIBRARY_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
+ament_target_dependencies(${LIBRARY_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+
+# Add and export library for composable node server
+add_library(${COMPOSABLE_NODE_NAME} SHARED src/servo_server.cpp)
+ament_target_dependencies(${COMPOSABLE_NODE_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(${COMPOSABLE_NODE_NAME} ${LIBRARY_NAME})
+rclcpp_components_register_nodes(${COMPOSABLE_NODE_NAME} "moveit_servo::ServoServer")
+
+# Add cpp interface example executable
+set(CPP_DEMO_NAME servo_demo)
+add_executable(${CPP_DEMO_NAME} src/cpp_interface_demo/servo_cpp_interface_demo.cpp)
+ament_target_dependencies(${CPP_DEMO_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(${CPP_DEMO_NAME} ${LIBRARY_NAME})
+
+# Add executable for using a controller
+set(SERVO_CONTROLLER_INPUT servo_controller_input)
+add_library(${SERVO_CONTROLLER_INPUT} SHARED src/teleop_demo/joystick_servo_example.cpp)
+ament_target_dependencies(${SERVO_CONTROLLER_INPUT} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+rclcpp_components_register_nodes(${SERVO_CONTROLLER_INPUT} "moveit_servo::JoyToServoPub")
+
+# Add executable to publish fake servo commands for testing/demo purposes
+set(FAKE_SERVO_CMDS_NAME fake_command_publisher)
+add_executable(${FAKE_SERVO_CMDS_NAME} test/publish_fake_jog_commands.cpp)
+ament_target_dependencies(${FAKE_SERVO_CMDS_NAME} 
+  rclcpp
+  geometry_msgs
+  std_srvs
 )
 
-################################################
-## An example of converting joystick commands ##
-################################################
-
-add_executable(spacenav_to_twist
-  src/teleop_examples/spacenav_to_twist.cpp
-)
-add_dependencies(spacenav_to_twist ${catkin_EXPORTED_TARGETS})
-target_link_libraries(spacenav_to_twist ${catkin_LIBRARIES})
-
-##################
-## INSTALLATION ##
-##################
-
-install(
-  TARGETS
-    ${LIBRARY_NAME}
-    servo_server
-    spacenav_to_twist
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+# Install Libraries
+install(TARGETS ${LIBRARY_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin)
+
+install(TARGETS
+  ${COMPOSABLE_NODE_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin)
+
+install(TARGETS
+  ${SERVO_CONTROLLER_INPUT}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin)
+
+install(TARGETS
+  ${CPP_DEMO_NAME}
+  DESTINATION lib/${PROJECT_NAME})
+
+install(TARGETS
+  ${FAKE_SERVO_CMDS_NAME}
+  DESTINATION lib/${PROJECT_NAME})
+
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+  find_package(ros_testing REQUIRED)
+  find_package(Boost REQUIRED COMPONENTS filesystem)
+
+  if(UNIX OR APPLE)
```

Comment:
Is macOS fully supported?

Reply:
I haven't done things there. If I am honest was copying a lot of the CMakeLists from an example. I can remove this

Reply:
Ok, I would remove it for now